composer.json

```diff
@@ -91,5 +104,23 @@
         }
     },
     "minimum-stability": "dev",
-    "prefer-stable": true
+    "prefer-stable": true,
+    "scripts": {
+        "post-install-cmd": [
+            "composer run-script post-update-cmd"
+        ],
+        "pre-update-cmd": [
+            "@putenv COMPOSER_MEMORY_LIMIT=2G"
```

Comment:
Like process-timeout, this also seems unnecessary and ineffective. The [default memory limit is 1.5GB](https://getcomposer.org/doc/articles/troubleshooting.md#memory-limit-errors), is an extra 0.5GB really a game-changer?

```
composer install --profile
Memory usage: 119.22MiB (peak: 1025.41MiB), time: 23.12s
```

It also doesn't do what I would expect. Setting it to something tiny (1M) has no effect on the install process except for the remove-git-submodules script.

Reply:
I think remove until there is a reason to change it. Another thing I borrowed that addresses a problem that may no longer exist. Maybe a composer 1 issue?